core/src/main/java/org/stellar/anchor/sep38/Sep38Service.java

```diff
@@ -248,6 +258,21 @@ public GetPriceResponse getPrice(Sep38GetPriceRequest getPriceRequest) throws An
         .build();
   }
 
+  private String getClientIdFromToken(Sep10Jwt token) throws AnchorException {
+    if (token == null || token.getAccount() == null) return null;
+    // Get clientId if the client exists
+    try {
+      GetCustomerResponse customer =
+          customerIntegration.getCustomer(
+              GetCustomerRequest.builder().account(token.getAccount()).build());
+      return customer.getId();
+    } catch (NotFoundException nfex) {
+      // skip only if customer not found.
+      debugF("Unable to get customer {} from customer integration server.", token.getAccount());
+    }
+    return null;
+  }
```

Comment:
I think there is a misunderstanding of what `clientId` is. Its not the ID of the customer requesting the quote.

From the documentation:

> Client IDs will be the Stellar public key the public key used to authenticate via SEP-10. Anchors must ensure that > the public key specified in the request matches a public key known to belong to the sending anchor

This is only true for SEP-31 though. In SEP-24 & SEP-6, the client ID is either the Stellar public key of the wallet OR the wallet's domain.

Reply:
@Ifropc wants to replace usage of Stellar public keys with domains, even for custodial applications. For 1.x, we probably need to continue using Stellar public keys, but for 2.x we could use home domains if we require the `clients.domain` field for each client.

Reply:
To clarify, not domains, but client name that's defined in the config

Reply:
From the spec
> An identifier for the client application making the request

So I believe using `client.name` is appropriate here, as initially the idea of introducing `client.name` was to use it as application identification

Reply:
Oh, yeah, I was wondering about this and forgot to check the documentation. 

I think the name `client_id` is very confusing though. Should we change to `client_account`? or `client_stellar_account`?

Reply:
I am not familiar with the context of the `client.name`. Can we add a Jira ticket for discussion?